scripts/Muon/GUI/Common/threading_manager.py

```diff
@@ -0,0 +1,282 @@
+from __future__ import (absolute_import, division, print_function)
+
+import traceback
+import sys
+
+from qtpy import QtCore, QtWidgets
+from qtpy.QtCore import Signal
+
+
+def split_list_into_n_parts(lst, n):
+    return [lst[i::n] for i in range(n)]
+
+
+def column(matrix, i):
+    return [row[i] for row in matrix]
+
+
+def split_kwarg_list(kwarg_list, n):
+    chunks = [split_list_into_n_parts(args, n) for args in kwarg_list.values()]
+    return [dict(zip(kwarg_list.keys(), column(chunks, i))) for i in range(n)]
+
+
+# Wraps a function, adding the following :
+# - Converts function arguments to allow for lists of arguments to be passed with the same names.
+# - Catches exceptions and returns a list of these as second return value.
+# - Adds a new argument 'progress_callback', to which user passes a signal, this is then emitting with
+# the progression of the evaluation of the argument list.
+def threading_decorator(fn):
+    def wrapper(progress_callback, **kwargs):
+        results, failed_results_exceptions = [], []
+        num_evals = len(kwargs.values()[0])
+        input_list = [{key: value[i] for key, value in kwargs.items()} for i in range(num_evals)]
+        for inputs in input_list:
+            try:
+                result = fn(**inputs)
+            except Exception as e:
+                failed_results_exceptions += [e]
+                continue
+            results += [result]
+            progress_callback.emit(100 / num_evals)
```

Comment:
```py
results += [result]
progress_callback.emit(100 / num_evals)
```
Could be moved inside the `try` block, thus removing the need for `continue`

Although it's a matter of style (and opinionated!), I think `.extend([e])` is more readable than `+= [e]` (and this occurs throughout the code).
`.append(e)` would probably be the most readable choice because you're only adding one element to the list at a time.